packages/framework-core/src/booster-events-reader.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  EventSearchRequest,
+  BoosterConfig,
+  Logger,
+  NotAuthorizedError,
+  EventSearchResponse,
+  NotFoundError,
+  EntityMetadata,
+  InvalidParameterError,
+  EventFilter,
+  EventFilterByEntity,
+  EventFilterByType,
+} from '@boostercloud/framework-types'
+import { BoosterAuth } from './booster-auth'
+import { Booster } from './booster'
+
+export class BoosterEventsReader {
+  public constructor(readonly config: BoosterConfig, readonly logger: Logger) {}
+
+  public async fetch(eventRequest: EventSearchRequest): Promise<Array<EventSearchResponse>> {
+    this.validateRequest(eventRequest)
+    return this.processFetch(eventRequest)
+  }
+
+  private validateRequest(eventRequest: EventSearchRequest): void {
+    this.logger.debug('Validating the following event request: ', eventRequest)
+    const entityMetadata = this.entityMetadataFromRequest(eventRequest)
+
+    if (!BoosterAuth.isUserAuthorized(entityMetadata.authorizeReadEvents, eventRequest.currentUser)) {
+      throw new NotAuthorizedError('Access denied for reading events')
+    }
+  }
+
+  private entityMetadataFromRequest(eventRequest: EventSearchRequest): EntityMetadata {
+    const { filters } = eventRequest
+    if (isByEntitySearch(filters) === isByEventTypeSearch(filters)) {
+      // Means no search type was passed or both were passed
```

Comment:
Do you mean to provide a better error message? Well, I did this to avoid cluttering too much the function code and used the message `"It should contain either "type" or "entity" field"`, that works well for when you didn't include any type or both types.

Would you prefer to throw two different errors in this case (including an extra if)?

Reply:
I'd say yeah, it'd be clearer for the user.

Reply:
Ok, I'll change it

Reply:
Done here: 081e964be027caedadfbfab396bf018645022701